javascript/ql/src/semmle/javascript/security/dataflow/TaintedPath.qll

```diff
@@ -93,13 +93,38 @@ module TaintedPath {
         |
           name = argumentlessMethodName
         )
-        or
+      )
+      or
+      // array method calls of interest
+      exists(DataFlow::MethodCallNode mcn, string name | dst = mcn and mcn.calls(src, name) |
+        // A `str.split()` call can either split into path elements (`str.split("/")`) or split by some other string.
         name = "split" and
-        not exists(DataFlow::Node splitBy | splitBy = mcn.getArgument(0) |
-          splitBy.mayHaveStringValue("/") or
-          any(DataFlow::RegExpLiteralNode reg | reg.getRoot().getAMatchedString() = "/")
-              .flowsTo(splitBy)
+        (
+          if
+            exists(DataFlow::Node splitBy | splitBy = mcn.getArgument(0) |
+              splitBy.mayHaveStringValue("/") or
+              any(DataFlow::RegExpLiteralNode reg | reg.getRoot().getAMatchedString() = "/")
+                  .flowsTo(splitBy)
```

Comment:
Any reason not to use `RegExpCreationNode`?

Reply:
No.  
I just reused the existing code for detecting `.split("/")`.